src/SourceMap.js

```diff
@@ -0,0 +1,192 @@
+const sourceMap = require('source-map');
+const textUtils = require('./utils/textUtils');
+
+class SourceMap {
+  constructor(file) {
+    this.sources = {};
+    this.mappings = [];
+    this.file = file;
+  }
+
+  copyConstructor(map) {
+    let sourcemap = new SourceMap();
+    sourcemap.mappings = map.mappings;
+    sourcemap.sources = map.sources;
+    sourcemap.file = map.file;
+    return sourcemap;
+  }
+
+  isConsumer(map) {
+    return map && map.computeColumnSpans;
```

Comment:
Can we just do `map instanceof sourceMap.SourceMapConsumer`? This check currently seems somewhat brittle since they could change the internal implementation.